### PR TITLE
Update setting_up_git.md

### LINKED ISSUE #28215: Setting up Git: A partition between two topics
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

There is a section in setting up git lesson where a certain section is given for mac users only and the next part is just normal tutorial for setting up git, but it causes problem cause there is no partition between the mac user special tut and normal guide which makes newbies think the next part is also part of mac users Tutorial, which causes huge problem in setting up git

in the image provided there is mac user specfic command and in next line i.e (Step 2.3) many linux user skip this thinking its related to mac user only,which result in them making no SSH key

ss

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-setting-up-git

### (Optional) Discord Name

Ashking

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by theSamyak**
issue #28215
In the "Setting Up Git" lesson (setting_up_git.md), added a note section to clearly differentiate macOS-specific commands from general setup instructions.

## Because

Added a note section in the "Setting Up Git" lesson to clearly differentiate macOS-specific commands from general setup instructions.

## This PR

Added a note section in setting_up_git.md to highlight macOS-specific commands and also changed the title "macOS Users" to "For macOS Users" improving clarity for users.

## Issue

Closes #28215

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [X] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Review by xandora — CHANGES_REQUESTED** [review]
We have specific styling code for these types of note blocks. Just using `>` is not the right way to do it in this situation.

Please review the style guide and update this PR accordingly before we can consider merging it. 

https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#note-boxes

File: foundations/installations/setting_up_git.md [hunk 1]
```diff
@@ -170,12 +170,17 @@ git config --get user.name
 git config --get user.email
 ```
 
-**macOS Users:** Run these two commands to tell Git to ignore .DS_Store files, which are automatically created when you use Finder to look into a folder. .DS_Store files are invisible to the user and hold custom attributes or metadata (like thumbnails) for the folder, and if you don't configure Git to ignore them, pesky .DS_Store files will show up in your commits. Remember to copy and paste each of these commands into your terminal.
+<div class="lesson-note" markdown="1">
+
```
Comment (xandora) [inline foundations/installations/setting_up_git.md:174]: Needs to be a blank line, currently there is one space.
```suggestion
 
```

File: foundations/installations/setting_up_git.md [hunk 2]
```diff
@@ -170,12 +170,17 @@ git config --get user.name
...
+  
+#### For macOS Users
+
+Run these two commands to tell Git to ignore .DS_Store files, which are automatically created when you use Finder to look into a folder. .DS_Store files are invisible to the user and hold custom attributes or metadata (like thumbnails) for the folder, and if you don't configure Git to ignore them, pesky .DS_Store files will show up in your commits. Remember to copy and paste each of these commands into your terminal.
 
 ```bash
 echo .DS_Store >> ~/.gitignore_global
 git config --global core.excludesfile ~/.gitignore_global
 ```
+</div>
```
Comment (xandora) [inline foundations/installations/setting_up_git.md:183]: There needs to be a blank line between the end of the code fence and the next element.
```suggestion

</div>
```

**Review by xandora — CHANGES_REQUESTED** [review]
@theSamyak Thanks for working on the previous suggestions, just a couple of minor touches needed to get this ready to go.

**Comment by theSamyak** [comment]
> @theSamyak Thanks for working on the previous suggestions, just a couple of minor touches needed to get this ready to go.

Hi, I messed up my commit history and there is still some linting issue and I apologize for it.  I have cloned the project into VS Code and from now on, will make the changes and contributions from there to maintain a cleaner commit history and preserve my sanity. Thanks.

**Review by xandora — APPROVED** [review]
Looks good to me!

**Pull request merged**